core/core_integrations/discord.js

```diff
@@ -108,73 +59,79 @@ let discord = require('discord.js'),
         return message;
     },
 
-    sendSingleMessage = function(message, threadId, channel, callback) {
-        stopTyping(threadId);
-        bot.sendMessage(channel, message, callback);
-    },
-
-    sendMultipleMessages = function(messageList, threadId, channel) {
-        if (!messageList || messageList.length <= 0) {
-            return;
+    sendMessage = (message, threadId) => {
+        let channel = lookUpChannel(threadId);
+        message = addMentions(message);
+        let messages = shim._chunkMessage(message, 2000);
+        for (let message of messages) {
+            channel.stopTyping();
+            channel.sendMessage(message);
         }
-        let message = messageList[0];
-        messageList.splice(0, 1);
-        sendSingleMessage(message, threadId, channel, sendMultipleMessages.bind(this, messageList, threadId, channel));
     },
 
-    sendMessage = function(message, threadId) {
+    sendFile = (type, file, description, threadId) => {
         let channel = lookUpChannel(threadId);
-        message = addMentions(message);
-        message = sentenceSplitter(message);
-        sendMultipleMessages(message, threadId, channel);
+        channel.stopTyping();
+        switch (type) {
+        case 'file':
+        case 'url':
+            channel.sendFile(file, '', description);
+            break;
```

Comment:
Yes this has been tested. According to Discord.js api sendFile supports Strings and Buffers. https://hydrabolt.github.io/discord.js/#!/docs/tag/master/typedef/FileResolvable

Changed logic to if-else;